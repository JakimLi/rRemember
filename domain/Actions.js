const types = {
  GENERATE: 'GENERATE',
  SHUFFLE: 'SHUFFLE',
  CHECK: 'CHECK'
};

const generate = () => {
  return {
    type: types.GENERATE
  };
};

const shuffle = () => {
  return {
    type: types.SHUFFLE
  };
};

const check = () => {
  return {
    type: types.CHECK
  };
};


export {generate, shuffle, check, types};
