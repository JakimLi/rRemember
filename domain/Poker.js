const colors = ['Spades', 'Hearts', 'Diamonds', 'Clubs'];
const points = ['A', '2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K'];
const jokers = ['BJ', 'LJ'];

const generate = () => {
  var cards = new Array();

  for(const color of colors) {
    for (const point of points) {
      cards.push({'color': color, 'point': point});
    }
  }

  for(const joker of jokers) {
    cards.push({'color': joker, 'point': '0'});
  }

  return cards;
};

const shuffle = (cards) => {
  return cards.concat().sort(() => 0.5 - Math.random());
}

export {generate, shuffle};
