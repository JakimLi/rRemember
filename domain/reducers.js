import { generate, shuffle } from './Poker';
import { types } from './actions';

export default (state = {}, action) => {
  switch(action.type) {
    case types.GENERATE:
      return {
        remembering: generate(),
        status: 'ready'
      };
    case types.SHUFFLE:
      return {
        remembering: shuffle(state.remembering),
        status: 'remembering'
      };
    case types.CHECK:
      return {
        remembering: state.remembering,
        status: 'checking'
      };
    default:
      return state;
  }
}
