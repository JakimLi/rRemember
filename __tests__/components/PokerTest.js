import 'react-native';
import React from 'react';
import PokerContainer from '../../components/Poker.js';
import { createStore } from 'redux';
import reducer from '../../domain/reducers';
import { generate } from '../../domain/actions';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';

let store = createStore(reducer);
store.dispatch(generate());

it('it renders correctly', () => {
  const poker = renderer.create(
    <Provider store={store}>
      <PokerContainer />
    </Provider>
  )

  expect(poker).toMatchSnapshot();
});
