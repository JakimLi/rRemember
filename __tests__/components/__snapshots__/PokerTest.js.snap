exports[`test it renders correctly 1`] = `
<ScrollView
  horizontal={true}
  pageEnbaled={true}
  style={
    Object {
      "backgroundColor": "#1a2229",
      "paddingTop": 100,
    }
  }>
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={1} />
  <Image
    source={undefined} />
  <Image
    source={undefined} />
</ScrollView>
`;
